File: url_redirect.py
import requests

def get_final_destination(url):
    response = requests.get(url, allow_redirects=True)
    return response.url

# Example usage
url = 'http://individual.troweprice.com/public/Retail/xStaticFiles/FormsAndLiterature/CollegeSavings/trp529Disclosure.pdf'
final_destination = get_final_destination(url)
print(final_destination)

File: url_redirect.ts
import axios, { AxiosResponse } from 'axios';

async function getFinalDestination(url: string): Promise<string> {
  const response = await axios.get(url, { maxRedirects: 15, validateStatus: null });
  return response.request.res.responseUrl || url;
}

// Example usage
const url = 'http://individual.troweprice.com/public/Retail/xStaticFiles/FormsAndLiterature/CollegeSavings/trp529Disclosure.pdf';
getFinalDestination(url)
  .then(finalDestination => console.log(finalDestination))
  .catch(error => console.error('Error:', error));
